examples/webgl_texture_raycast.html

```diff
@@ -0,0 +1,377 @@
+						this._parentTexture[i].needsUpdate = true;
+
+					}
+
+				}
+
+			}
+
+		</script>
+		<script>
+
+			var width = window.innerWidth;
+			var height = window.innerHeight;
+
+			var canvas;
+			var objects = [];
+			var planeTexture, cubeTexture, circleTexture;
+
+			var container;
+
+			var camera, scene, renderer;
+
+			var raycaster = new THREE.Raycaster();
+			var mouse = new THREE.Vector2();
+			var onClickPosition = new THREE.Vector2();
+
+			init();
+			render();
+			
+			function init() {
+
+				container = document.getElementById("container");
+
+				scene = new THREE.Scene();
+
+				camera = new THREE.PerspectiveCamera(45, width / height, 1, 1000);
+				camera.position.x = -30;
+				camera.position.y = 40;
+				camera.position.z = 50;
+				camera.lookAt(scene.position);
+			
+				renderer = new THREE.WebGLRenderer();
+				renderer.setClearColor(new THREE.Color(0xEEEEEE, 1.0));
+				renderer.setSize(width, height);
+				container.appendChild(renderer.domElement);
+
+				// A cube, in the middle.
+				cubeTexture = new THREE.Texture(undefined, THREE.UVMapping, THREE.RepeatWrapping, THREE.RepeatWrapping);
+				canvas = new CanvasTexture(cubeTexture);
+				var cubeMaterial = new THREE.MeshBasicMaterial({ map: cubeTexture });
+				var cubeGeometry = new THREE.BoxGeometry(20, 20, 20);
+				// Set a specific texture mapping.
+				var uvs;
+				for (var i = 0; i < cubeGeometry.faceVertexUvs[0].length; i++) {
+
+					uvs = cubeGeometry.faceVertexUvs[0][i];
+					for (var j = 0; j < 3; j++) {
+
+						if (uvs[j].x < 0.1) uvs[j].x = -1;
+						if (uvs[j].y < 0.1) uvs[j].y = -1;
+
+					}
+
+				}
+				var cube = new THREE.Mesh(cubeGeometry, cubeMaterial);
+				cube.position.x = 4;
+				cube.position.y = -5;
+				cube.position.z = 0;
+				objects.push(cube);
+				scene.add(cube);
+
+				// A plane on the left.
+				planeTexture = new THREE.Texture(undefined, THREE.UVMapping, THREE.MirroredRepeatWrapping, THREE.MirroredRepeatWrapping);
+				canvas.addParent(planeTexture);
+				var planeMaterial = new THREE.MeshBasicMaterial({ map: planeTexture });
+				var planeGeometry = new THREE.PlaneBufferGeometry(25, 25, 1, 1);
+				var uvs = planeGeometry.attributes.uv.array;
+				for (var i = 0; i < uvs.length; i++) {
+
+					uvs[i] *= 2;
+
+				}
+				var plane = new THREE.Mesh(planeGeometry, planeMaterial);
+				plane.position.x = -16;
+				plane.position.y = -5;
+				plane.position.z = 0;
+				objects.push(plane);
+				scene.add(plane);
+
+				// A circle on the right.
+				circleTexture = new THREE.Texture(undefined, THREE.UVMapping, THREE.ClampToEdgeWrapping, THREE.ClampToEdgeWrapping);
+				canvas.addParent(circleTexture);
+				var circleMaterial = new THREE.MeshBasicMaterial({ map: circleTexture });
+				var circleGeometry = new THREE.CircleBufferGeometry(25, 40, 0, Math.PI * 2);
+				var uvs = circleGeometry.attributes.uv.array;
+				for (var i = 0; i < uvs.length; i++) {
+
+					uvs[i] = (uvs[i] - 0.25) * 2;
```

Comment:
I will, but what do you means by "it does not look right"?
It is rendered correctly, and I think that the way I calculate the intersection is the right one. For me it works as I expected.
If you mean, the picture isn’t very pretty, I think you are right, but I always think ClampToEdgeWrapping does ugly picture. I just (try to) prove it works.